### ISSUE TITLE
Feature request: Switch to toggle HDR and High FPS modes

### ISSUE THREAD

**Issue by brun0san** [briis/unifiprotect#128]
Would it be possible to add a switch to toggle the HDR and High FPS modes on/off? The idea is to replicate the switches that have been implemented to toggle IR and Status Light on/off. This not only would make writing automations simpler but also allow for toggles to be added to the Lovelace interface so this can be easily enabled/disabled for each camera.

The recent services implemented in 0.5.8 are good to write automations but they don’t allow to get the current status/state of such modes from the cameras.

**Comment by briis** [comment]
Yes it is possible. Only issue is that when running on a CloudKey the High FPS will fail, so it requires a little more, NOT to enable that Switch when on CloudKey.

**Comment by brun0san** [comment]
Excellent! For now, I’d be happy with a switch for HDR as it’s what I’m using at the moment. I probably wouldn’t be able to test the switch for High FPS given the version I’m running still doesn’t have that feature.

**Comment by briis** [comment]
Perfect. I need someone to test where High FPS will fail. I'll work on it.
Are you running CloudKey or UDMP/UNVR?

**Comment by briis** [comment]
There is now a Beta Release (0.6.0-beta-2) which includes the new switches. I only create the HDR Switch for Non G4 Cameras, as this is not supported anymore with the latest release of Unifi Protect. And the HighFPS switches are then only created for the G4 cameras, as this is not supported on lower resolution cameras.

If you have the change to test it out, I would be grateful.

**Comment by brun0san** [comment]
Hi,
I’m using Cloudkey, but it looks like I won’t be able to test anybody yours changes. I’m currently using the HDR service with Protect v1.13.2, which supports HDR switch on/off for G4 cameras. HDR is only supported on 2 G4 cameras I have. Since still running on 1.13.2 none of my cameras support High FPS. Since you’ve implemented the HDR service call for G4 cameras, why not create the switch as well? 

As far as I can tell HDR on G4 cameras was disabled as of v1.13.4, so perhaps you can check the controller version to figure out whether to create switches/services?

**Comment by brun0san** [comment]
The latest Protect version is 1.13.7, I’m only running two minor releases behind as I only tend to update when stable versions are available, and even then only if worth the trouble as I’ve had problems with buggy versions before...

**Comment by briis** [comment]
OK. Let me see if I can do a check for either CloudKey or SW Version, and then enable the HDR feature.

**Comment by briis** [comment]
I have now modified the code, so that switches are only created if the Camera tells me it is available. I have tested this on a CloudKey V1.3.7, so could do me a favour and download the Beta 3 and test this out on your V1.3.2?

**Comment by brun0san** [comment]
Great job mate!

I can confirm the switch was automatically added upon restart and it works perfectly fine.

Just confirming my environment for your reference:

CloudKey Firmware: v.1.1.11
UniFi Protect: v.1.13.2
2 x G4 cameras

**Comment by brun0san** [comment]
Also, you’ve mentioned you’ve tested on v1.13.7, so I was just wondering for a potential future upgrade, are you able to control HDR on v1.13.7?

If not, it looks like I won’t be upgrading any time soon :)

**Comment by briis** [comment]
Thanks for the feedback. There is one downside to this approach: If Unifi Protect changes the way HDR can be controlled in the Future, you will have to manually delete the switch on the Integration page.

For your second question: I am not sure. During my testing I found that Unifi is changing the way they use HDR going forward. I did not test any G4 Cameras on the CloudKey, only two G3. The G3 Flex and the G3 Bullet. G3 Flex supports HDR but not the Bullet Camera.
On my UDMP I have 3 G4, two PRO and one Doorbell - None of those support HDR switching anymore - As per release notes, HDR is always on. They introduced WDR instead, but I have not played with adjusting that.

Hope this helps.
